### PR TITLE
backend/enh/first-pickup-route-call-to-google

### PR DESCRIPTION
**Pull request by rupakkorde**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added integration tests for my changes where possible
- [ ] I tested the changes using integration tests
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/rider-platform/rider-app/Main/src/Tools/Maps.hs [hunk 1]
```diff
@@ -105,11 +105,15 @@ getRoutes isAvoidToll personId merchantId mbMOCId req = do
...
   runWithServiceConfig (Maps.getRoutes $ fromMaybe merchant.isAvoidToll isAvoidToll) (.getRoutes) merchantId mOCId req
 
-getPickupRoutes :: ServiceFlow m r => Id Person -> Id Merchant -> Maybe (Id MerchantOperatingCity) -> GetRoutesReq -> m GetRoutesResp
-getPickupRoutes personId merchantId mbMOCId req = do
+getPickupRoutes :: ServiceFlow m r => Id Merchant -> Id MerchantOperatingCity -> MapsService -> GetRoutesReq -> m GetRoutesResp
+getPickupRoutes merchantId merchantOperatingCityId service req = do
   merchant <- SMerchant.findById merchantId >>= fromMaybeM (MerchantNotFound merchantId.getId)
-  mOCId <- getMerchantOperatingCityId personId mbMOCId
-  runWithServiceConfig (Maps.getRoutes merchant.isAvoidToll) (.getPickupRoutes) merchantId mOCId req
+  merchantMapsServiceConfig <-
```
Comment (khuzema786) [inline Backend/app/rider-platform/rider-app/Main/src/Tools/Maps.hs:111]: Can `runWithServiceConfig` function be used here with (.getPickupRoutes) or (.getFirstPickupRoute) changed accordingly ?

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Route.hs [hunk 2]
```diff
@@ -18,24 +18,54 @@ module Domain.Action.UI.Route
...
 getRoutes :: (ServiceFlow m r, EsqDBReplicaFlow m r) => (Id DP.Person, Id Merchant.Merchant) -> Maps.GetRoutesReq -> m Maps.GetRoutesResp
 getRoutes (personId, merchantId) req = do
   Maps.getRoutes Nothing personId merchantId Nothing req
 
-getPickupRoutes :: ServiceFlow m r => (Id DP.Person, Id Merchant.Merchant) -> Maps.GetRoutesReq -> m Maps.GetRoutesResp
-getPickupRoutes (personId, merchantId) req = do
-  Maps.getPickupRoutes personId merchantId Nothing req
+getPickupRoutes :: (Id DP.Person, Id Merchant.Merchant) -> GetPickupRoutesReq -> Flow Maps.GetRoutesResp
+getPickupRoutes (personId, merchantId) GetPickupRoutesReq {..} = do
+  mocId <- Maps.getMerchantOperatingCityId personId Nothing
```
Comment (khuzema786) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Route.hs:53]: Can runWithServiceConfig function be used here with (.getPickupRoutes) or (.getFirstPickupRoute) changed accordingly ?

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Ride.hs [hunk 3]
```diff
@@ -267,7 +267,7 @@ buildRide driver booking ghrId otp enableFrequentLocationUpdates clientId previo
         estimatedEndTimeRange = Nothing,
         rideTags = Nothing,
         hasStops = booking.hasStops,
-        isPickupOrDestinationEdited = Nothing
```
Comment (khuzema786) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Ride.hs:270]: What is this change ?

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Ride.hs [hunk 4]
```diff
@@ -267,7 +267,7 @@ buildRide driver booking ghrId otp enableFrequentLocationUpdates clientId previo
...
+        isPickupOrDestinationEdited = Just False
```
Comment (khuzema786) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Ride.hs:270]: What is this change ?

File: Backend/app/rider-platform/rider-app/Main/src/Tools/Maps.hs (hunk 1 above)
Comment (rupakkorde) [reply Backend/app/rider-platform/rider-app/Main/src/Tools/Maps.hs:111]: We have set getFirstPickupRoute as Maybe Service so we can set it as NULL to disable query
So can't pass it

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Ride.hs (hunk 4 above)
Comment (rupakkorde) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Ride.hs:270]: Incase null case is not handled in json logic for any Maybe field, jsonlogic gives null
That's why added just for precaution

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Route.hs (hunk 2 above)
Comment (rupakkorde) [reply Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Route.hs:53]: Same as above

**Pull request merged**